src/modules/index.ts

```diff
@@ -0,0 +1,19 @@
+import {isAndroid} from '../utils/platform';
+
+import type {AmazonModuleProps} from './amazon';
+import {AndroidModule, AndroidModuleProps} from './android';
+import {IosModule, IosModuleProps} from './ios';
+
+declare module 'react-native' {
+  interface NativeModulesStatic {
+    RNIapIos: IosModuleProps;
+    RNIapModule: AndroidModuleProps;
+    RNIapAmazonModule: AmazonModuleProps;
+  }
+}
+
+export * from './amazon';
+export * from './android';
+export * from './ios';
+
+export const NativeModule = isAndroid ? AndroidModule : IosModule;
```

Comment:
Amazon missing here?

Reply:
Amazon is checked [here](https://github.com/dooboolab/react-native-iap/pull/1797/files/beca8d361c6aedd4a19efc21dbecd4ceffe2a198#diff-32fc8165c1fa57159a39b1ef5a3c01c779827b25c9513f25360dc66cde072f8cR88). It's either iOS or Android as "device" modules, and Android can either be Play Store or Amazon IAP.